### PR TITLE
[lint] fix github actions for forked PRs

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26562 [lint] fix github actions for forked PRs**

I was trying to be too clever with GITHUB_HEAD_REF...

Differential Revision: [D17538517](https://our.internmc.facebook.com/intern/diff/D17538517)

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -24,9 +24,11 @@ jobs:
             # We are on master, just set the SHA from our current location
             echo ::set-output name=commit_sha::${GITHUB_SHA}
           else
-            # We are on a PR, we need to check out PR branch
-            git checkout ${GITHUB_HEAD_REF}
-            echo ::set-output name=commit_sha::$(git rev-parse ${GITHUB_HEAD_REF})
+            # We are on a PR, so actions/checkout leaves us on merge commit.
+            # Check out the actual tip of the branch.
+            PR_TIP=$(git rev-parse HEAD^2)
```
Comment (ezyang) [inline .github/workflows/lint.yml]: This seems kind of fragile to me though

Comment (suo) [reply .github/workflows/lint.yml]: I agree, but it's was what Ed Thompson suggested as the right way to do it—I think it's supposed to be a documented fact that PRs check out to the merge ref (merge A into B), so it *should* be stable rely on the second parent being A.

**Comment by eellison** [comment]
I'm getting an error in https://github.com/pytorch/pytorch/pull/26683 with 

> + '[' -z resolve_ignored_mod_types ']' + git checkout resolve_ignored_mod_types error: pathspec 'resolve_ignored_mod_types' did not match any file(s) known to git ##[error]Process completed with exit code 1.

Is this related ?

**Comment by suo** [comment]
yes, this PR should fix it

**Pull request closed without merging**